openmdao/drivers/tests/test_pyoptsparse_driver.py

```diff
@@ -2047,6 +2049,32 @@ def test_ParOpt_basic(self):
         assert_near_equal(prob['z'][0], 1.9776, 1e-3)
         assert_near_equal(prob['obj_cmp.obj'][0], 3.183, 1e-3)
 
+    def test_snopt_analysis_error(self):
+
+        x_tr = np.linspace(0, 2*np.pi, 100)
+        y_tr = np.sin(x_tr)
+
+        p = om.Problem(model=om.Group())
+
+        p.driver = om.pyOptSparseDriver(optimizer='SNOPT')
+
+        mm = om.MetaModelStructuredComp(extrapolate=False)
+        mm.add_input('x', val=1.0, training_data=x_tr)
+        mm.add_output('y', val=1.0, training_data=y_tr)
+        p.model.add_subsystem('interp', mm, promotes_inputs=['x'], promotes_outputs=['y'])
+
+        p.model.add_objective('y', scaler=-1)
+        p.model.add_design_var('x', lower=6, upper=10)
+
+        p.set_solver_print(level=0)
+        p.setup()
+
+        p.set_val('x', 0.75)
+
+        msg = "Analysis Error: Line 205 of file {}".format(inspect.getsourcefile(MetaModelStructuredComp))
```

Comment:
This test should be in the metamodel tests, rather than in pyoptsparse/snopt.

Reply:
fixed